### ISSUE TITLE
Problem detecting DrupalRoot on Drupal 8.7

### ISSUE THREAD

**Issue by Etroid** [webflo/drupal-finder#46]
DrupalFinder is unable to detect drupalRoot is `/web` when using drupal core 8.7 - using `wikimedia/composer-merge-plugin`. Instead, it thinks the drupalRoot is `/`.

This issue occurs when running any `drush` or `drupal` command.

```
{
  "require": {
    "drupal/console": "~1.0",
    "drupal/core": "^8.7",
    "drush/drush": "^9.2"
  },
  "extra": {
    "installer-paths": {
      "web/core": [
        "type:drupal-core"
      ],
      "web/modules/contrib/{$name}": [
        "type:drupal-module"
      ],
      "web/profiles/contrib/{$name}": [
        "type:drupal-profile"
      ],
      "web/themes/contrib/{$name}": [
        "type:drupal-theme"
      ],
      "drush/contrib/{$name}": [
        "type:drupal-drush"
      ]
    }
  }
}

```

From what I gather this is because of a recent change to support Drupal core 8.8 (i.e. https://www.drupal.org/project/drupal/issues/2912387#comment-13188592)

**Comment by webflo** [comment]
Did you specify a name property in your composer.json?

**Comment by Etroid** [comment]
yes, the snippet is just part of the entire composer.json file All was working last week, a few days before the latest release of drupal-finder

**Comment by simesy** [comment]
Check if your main project `composer.json` is set to:
```
{
    "name": "drupal/drupal",
    ...
 ```
I had an old project I inherited that must have been initially modelled on the drupal composer.json. The update on Aug 2 [added logic](https://github.com/webflo/drupal-finder/blob/123e248e14ee8dd3fbe89fb5a733a6cf91f5820e/src/DrupalFinder.php#L118) that looks for `drupal/drupal` and sets the Drupal Root incorrectly.

**Comment by simesy** [comment]
Something to note, this problem didn't occur for me when the entry point was Drush launcher. It only occured when i was explicitly doing like `../vendor/bin/drush status` from the Drupal root.

**Comment by Etroid** [comment]
^^ @simesy correct, I was not using drush launcher when I encountered the problem. It was working before the latest release however.

**Comment by simesy** [comment]
@Etroid you haven't verified what that value of "name" is in your composer.json.

**Comment by Etroid** [comment]
@simesy, name is `"name": "tableau/www"` so should not be related.

**Comment by Techwolf12** [comment]
Same issue here, using the lightning distribution.
`"name": "drupal/drupal",`

Except the composer.json isn't in the docroot, but one directory outside of it for security reasons.

**Comment by simesy** [comment]
@Techwolf12 yeah that is pretty standard, drupal-finder is designed start in the docroot and look up a directory for the composer file.

`"name": "drupal/drupal",` is arguably incorrect because it collides with the well known composer package `drupal/drupal`, see attached image as evidence that Composer assumes this would never happen. It works because you're likely requiring bringing in the composer-friendly variation `drupal/core`.

I would change it. I've never seen negative consequences of changing it.

<img width="708" alt="Screen Shot 2019-09-24 at 9 49 04 AM" src="https://user-images.githubusercontent.com/188856/65471072-21d9a480-deb1-11e9-8365-a890e013c1bb.png">

**Comment by pjcdawkins** [comment]
Or remove it, there's no need for a `"name"` in a not-published Composer project.
